src/MongoDB/Session.c

```diff
@@ -343,6 +368,26 @@ static PHP_METHOD(Session, getTransactionOptions)
 	}
 } /* }}} */
 
+/* {{{ proto string MongoDB\Driver\Session::getTransactionState()
+   Returns the current transaction state for this session */
+static PHP_METHOD(Session, getTransactionState)
+{
+	php_phongo_session_t* intern;
+	mongoc_transaction_state_t state;
+
+	intern = Z_SESSION_OBJ_P(getThis());
+	SESSION_CHECK_LIVELINESS(intern, "getTransactionState")
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	state = mongoc_client_session_get_transaction_state(intern->client_session);
+
+	PHONGO_RETURN_STRING(php_phongo_get_transaction_state_string(state TSRMLS_CC));
```

Comment:
Similar to my concerns in https://github.com/mongodb/mongo-php-driver/pull/1062#discussion_r353810821, I think we should only call `PHONGO_RETURN_STRING` if `php_phongo_get_transaction_state_string` returns a non-null value; otherwise we risk passing `null` to `strlen()`. Consider:

```c
const char *state;

// ...

// Or keep state and add a state_str var if this is too long for your liking
state = php_phongo_get_transaction_state_string(mongoc_client_session_get_transaction_state(intern->client_session) TSRMLS_CC);

if (!state) {
    /* Exception already thrown */
    return;
}

PHONGO_RETURN_STRING(state);
```

We won't be able to write a test for this (you can do so manually if you like), but it provides some peace of mind.

Reply:
Logic added. The joy of working with pointers :)